server/me.go

```diff
@@ -66,22 +77,32 @@ func (s *Service) Me(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	email, err := getEmail(ctx)
+	username, err := getUsername(ctx)
+	if err != nil {
+		invalidData(w, err, s.Logger)
+		return
+	}
+	provider, err := getProvider(ctx)
 	if err != nil {
 		invalidData(w, err, s.Logger)
 		return
 	}
 
-	usr, err := s.UsersStore.Get(ctx, email)
-	if err == nil {
+	usr, err := s.UsersStore.Get(ctx, chronograf.UserQuery{Name: &username, Provider: &provider})
+	if err != nil && err != chronograf.ErrUserNotFound {
+		unknownErrorWithMessage(w, err, s.Logger)
+		return
+	}
+
+	if usr != nil {
 		res := newMeResponse(usr)
 		encodeJSON(w, http.StatusOK, res, s.Logger)
 		return
 	}
 
 	// Because we didnt find a user, making a new one
 	user := &chronograf.User{
-		Name: email,
+		Name: username,
```

Comment:
when creating a new user, we need to store both:
1. the `Provider`
1. the `Scheme`
we don't currently have the scheme on context, i believe, so this may be something we have to figure out per authentication provider response

Reply:
Or at least for now we should prob put OAuth2 for future-proofing / later migration purposes.

Reply:
@jaredscheib the principal has an issuer... I've imagined that could be the provider:
https://github.com/influxdata/chronograf/blob/master/oauth2/oauth2.go#L34

Additionally, I think another key field should be added to the principal of "Scheme."  As you point out right now it would be oauth2

Does that work?

Reply:
@goller should we store that as an `amr` claim on the jwt? 

http://openid.net/specs/openid-connect-core-1_0.html

Reply:
I'm ok with amr... the standard leaves it open to other stings.

Here are the currently registered strings: https://tools.ietf.org/html/draft-ietf-oauth-amr-values-04#section-2

Reply:
the Provider is the same as Issuer, yes (e.g. `GitHub`, etc.); Scheme is something else (like `OAuth2`, `LDAP`, etc.)